compiler/rustc_hir_analysis/src/check/compare_impl_item.rs

```diff
@@ -2026,10 +2026,19 @@ pub(super) fn check_type_bounds<'tcx>(
     // to its definition type. This should be the param-env we use to *prove* the
     // predicate too, but we don't do that because of performance issues.
     // See <https://github.com/rust-lang/rust/pull/117542#issue-1976337685>.
+    let trait_projection_ty = Ty::new_projection(tcx, trait_ty.def_id, rebased_args);
+    let impl_identity_ty = tcx.type_of(impl_ty.def_id).instantiate_identity();
```

Comment:
move this into the `next_solver` branch?

Reply:
then it happens once per `for mut obligation in util::elaborate(tcx, obligations) {`.

I could invert the `if` statement, and duplicate the for loop?

Reply:
you're right, didn't see the for loop :+1: